tests/sles4sap/publiccloud/qesap_terraform.pm

```diff
@@ -112,30 +112,39 @@ sub run {
     # variable to hold the top level container where the PTF directory
     # is going to be
     my $ptf_container;
+    my $hana_token;
+    my $ptf_token;
 
     if (get_var('HANA_ACCOUNT') && get_var('HANA_CONTAINER') && get_var('HANA_KEYNAME')) {
-        my $escaped_token = qesap_az_create_sas_token(
+        my $hana_token = qesap_az_create_sas_token(
             storage => get_required_var('HANA_ACCOUNT'),
             container => (split("/", get_required_var('HANA_CONTAINER')))[0],
             keyname => get_required_var('HANA_KEYNAME'),
             # lifetime has to be enough to reach the point of the test that
             # executes qe-sap-deployment Ansible playbook 'sap-hana-download-media.yaml'
             lifetime => 90,
-            permission => get_var('HANA_TOKEN_PERMISSION', 'r'));
+            permission => 'r');
         # escape needed by 'sed'
         # but not implemented in file_content_replace() yet poo#120690
-        $escaped_token =~ s/\&/\\\&/g;
-        set_var("HANA_TOKEN", $escaped_token);
-
-        if (get_var('PTF_INSTALL')) {
-            $ptf_files = qesap_az_list_container_files(
-                storage => get_required_var('HANA_ACCOUNT'),
-                container => (split("/", get_required_var('HANA_CONTAINER')))[0],
-                token => get_required_var('HANA_TOKEN'),
-                prefix => 'ptf'
-            );
-            $ptf_container = (split("/", get_required_var('HANA_CONTAINER')))[0];
-        }
+        my $escaped_hana_token;
+        ($escaped_hana_token = $hana_token) =~ s/\&/\\\&/g;
```

Comment:
Is the `$escaped_hana_token` variable required? I see it only used in the `set_var` call and `$hana_token' is not used anymore as well.
So the first option would be: 
```
$hana_token =~  s/\&/\\\&/g;
set_var("HANA_TOKEN", $hana_token);
```

You can as well declare new variable and substitute in one line:
```
my $hana_token = 'as=654321-654%ASD&sdfsdfsd&%xfsdfsf65421-'; 
(my $escaped_hana_token = $hana_token) =~ s/\&/\\\&/g; 
print "Hana token: $hana_token\nEscapend token: $escaped_hana_token\n"; 

Result:
Hana token: as=654321-654%ASD&sdfsdfsd&%xfsdfsf65421-
Escapend token: as=654321-654%ASD\&sdfsdfsd\&%xfsdfsf65421-
```
And there is as well the non-destructive substitution `s///gr`:
```
my $hana_token = 'as=654321-654%ASD&sdfsdfsd&%xfsdfsf65421-'; 
my $escaped_hana_token = $hana_token =~ s/\&/\\\&/gr; 
print "Hana token: $hana_token\nEscapend token: $escaped_hana_token\n";

Result:
Hana token: as=654321-654%ASD&sdfsdfsd&%xfsdfsf65421-
Escapend token: as=654321-654%ASD\&sdfsdfsd\&%xfsdfsf65421-
```